stdlib/string.gr

```diff
@@ -1609,3 +1609,64 @@ export let rec forEachCodePointi = (fn: (Number, Number) -> Void, str: String) =
   Memory.decRef(WasmI32.fromGrain(forEachCodePointi))
   void
 }
+
+/**
+ * Trims a string removing any leading or following white spaces based on the end param.
+ *
+ * @param str: The string to be trimmed.
+ * @param end: If true we are trimming the following spaces, otherwise trim leading spaces.
+ * @returns The trimmed string.
+ * 
+ * @example String.trimString("   Hello World   ", false)
+ */
+let trimString = (str: String, end: Bool) => {
+  let chars = explode(str), charsLength = length(str)
+  let mut i = 0, offset = 1
+  if (end) {
+    i = charsLength-1
+    offset = -1
+  }
+  for (; i < charsLength && i > -1; i += offset) {
+    if (
+      // Spacing
+      chars[i] != '	'  && // Tab
+      chars[i] != ''  && // LINE TABULATION
```

Comment:
Maybe we should use `\u` escapes for these constants? Also, there should be a TODO to use the Unicode whitespace property once #661 is completed.

Reply:
Fixed in latest commit.